src/scraper.py

```diff
@@ -0,0 +1,102 @@
+from contextlib import contextmanager
```

Comment:
A docstring at the top very shortly summarizing what a file does is in my experience very valuable for understanding the code in that file.

Reply:
Done